### ISSUE TITLE
No way to silence a new lint across both toolchains

### ISSUE THREAD

**Issue by RReverser** [rust-lang/rust-clippy#6860]
I'm trying to allow a single nightly-only Clippy lint.

```rust
#![warn(clippy::pedantic)]
#![allow(clippy::missing_panics_doc)]
```

When my code is compiled on nightly, everything is fine, but on stable (Rust 1.50) this results in:

```
warning: unknown clippy lint: clippy::missing_panics_doc
  --> src\lib.rs:25:5
   |
25 |     clippy::missing_panics_doc
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `clippy::missing_errors_doc`
   |
   = note: `#[warn(clippy::unknown_clippy_lints)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unknown_clippy_lints

warning: 1 warning emitted
```

Ok, I go ahead and add `#[allow(clippy::unknown_clippy_lints)]` using the lint name taken from the suggestion:

```rust
#![warn(clippy::pedantic)]
#![allow(clippy::unknown_clippy_lints, clippy::missing_panics_doc)]
```

Now the roles swap and my code compiles without warnings on stable but produces a warning on nightly:

```
warning: lint `clippy::unknown_clippy_lints` has been removed: this lint has been integrated into the `unknown_lints` rustc lint      
  --> src\lib.rs:22:5
   |
22 |     clippy::unknown_clippy_lints,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 2 warnings emitted
```

Ok, I go ahead and change `#[allow(clippy::unknown_clippy_lints)]` I just added above to ``#[allow(unknown_lints)]`:

```
#![warn(clippy::pedantic)]
#![allow(unknown_lints, clippy::missing_panics_doc)]
```

Roles swap again, and my code compiles without warnings on nightly but produces a warning on stable:

```
warning: unknown clippy lint: clippy::missing_panics_doc
  --> src\lib.rs:26:5
   |
26 |     clippy::missing_panics_doc
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^ help: did you mean: `clippy::missing_errors_doc`
   |
   = note: `#[warn(clippy::unknown_clippy_lints)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#unknown_clippy_lints

warning: 1 warning emitted
```

At this point I give up, because there seems to be no way to silence a lint that would work on both stable and nightly toolchains, short of adding and using a separate `feature = "nightly"` which seems like even more hassle.

Could Clippy perhaps disable the warning "warning: lint `clippy::unknown_clippy_lints` has been removed" and let user use it even on nightly until it's fully integrated and released on stable toolchain as well?

**Comment by giraffate** [comment]
It's because `unknown_clippy_lints` was deprecated from 1.51.0 (currently beta). For now, I may be wrong but can codes below help you?

```rust
#![warn(clippy::pedantic)]
#![allow(renamed_and_removed_lints, clippy::unknown_clippy_lints, clippy::missing_panics_doc)]

fn main() {}

/// This needs to be documented
pub fn unwrap() {
    let result = Err("Hi");
    result.unwrap()
}

```

**Comment by RReverser** [comment]
@giraffate Unfortunately, no - looks like `renamed_and_removed_lints` somehow ignores the `missing_panics_doc` (that is, the latter still gets a warning).

**Comment by RReverser** [comment]
Ah, no, sorry, all three together do work.

Could `renamed_and_removed_lints` be shown in warnings like the one above? That is, it could be shown as `note: `#[warn(renamed_and_removed_lints)]` on by default` on the corresponding warning.

**Comment by giraffate** [comment]
I found it during debugging based on your report.

I tried the codes below:
```
#![warn(clippy::pedantic)]
#![allow(clippy::unknown_clippy_lints, clippy::missing_panics_doc)]

fn main() {}

/// This needs to be documented
pub fn unwrap() {
    let result = Err("Hi");
    result.unwrap()
}

```
I got messages including `renamed_and_removed_lints`:
```
$ cargo +nightly -V    
cargo 1.52.0-nightly (970bc67c3 2021-03-07)
$ cargo +nightly clippy   
warning: lint `clippy::unknown_clippy_lints` has been removed: this lint has been integrated into the `unknown_lints` rustc lint
 --> src/main.rs:2:10
  |
2 | #![allow(clippy::unknown_clippy_lints, clippy::missing_panics_doc)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(renamed_and_removed_lints)]` on by default

warning: lint `clippy::unknown_clippy_lints` has been removed: this lint has been integrated into the `unknown_lints` rustc lint
 --> src/main.rs:2:10
  |
2 | #![allow(clippy::unknown_clippy_lints, clippy::missing_panics_doc)]
  |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

warning: 2 warnings emitted

    Finished dev [unoptimized + debuginfo] target(s) in 0.00s
```

**Comment by RReverser** [comment]
Huh, that's really odd... I guess I've missed it between the duplicate warnings. Thanks.
